pkg/agent/node/node.go

```diff
@@ -32,17 +32,15 @@ func Params() map[string]interface{} {
 		"requestedHostname": os.Getenv("CATTLE_NODE_NAME"),
 	}
 
-	for k, v := range params {
-		if m, ok := v.(map[string]string); ok {
-			for k, v := range m {
-				logrus.Infof("Option %s=%s", k, v)
-			}
-		} else {
-			logrus.Infof("Option %s=%v", k, v)
+	dclient, err := client.NewClientWithOpts(client.WithAPIVersionNegotiation())
+	if err != nil {
+		logrus.Errorf("Error getting client with API version negotiation, using default client from Env: %v", err)
+		dclient, err = client.NewClientWithOpts(client.FromEnv)
+		if err != nil {
+			logrus.Errorf("Error getting default client from Env: %v", err)
 		}
 	}
 
```

Comment:
Can we log the retrieved Docker info in trace so we can compare the local value with the remote value (if something goes wrong here)

Reply:
I moved the params logging (https://github.com/rancher/rancher/pull/32747/files#diff-93fe6e130b2e58fb15b3ec4070b80b9e71e37b443b225179c32dc7a1f1cd864dR52) after [setting dockerInfo in params](https://github.com/rancher/rancher/pull/32747/files#diff-93fe6e130b2e58fb15b3ec4070b80b9e71e37b443b225179c32dc7a1f1cd864dL50) so now it'll get logged similar to other options

Reply:
Can we use client.NewClientWithOpts(client.WithAPIVersionNegotiation(), client.FromEnv)? By doing this it won't break if user is using the env variables for their docker socket.

Reply:
Yeah @superseb and I did discussed that, but ugh now I don't remember why I skipped it. Adding FromEnv there as well